core/domain/blog_domain.py

```diff
@@ -348,7 +348,9 @@ def update_thumbnail_filename(
             new_thumbnail_filename: str|None. The updated thumbnail filename
                 for the blog post.
         """
-        self.require_valid_thumbnail_filename(self.thumbnail_filename) # type: ignore[arg-type]
+        if new_thumbnail_filename is not None:
+            if not isinstance(new_thumbnail_filename, str):
+                self.require_valid_thumbnail_filename(new_thumbnail_filename)
```

Comment:
This should be enough?
```suggestion
        if not isinstance(new_thumbnail_filename, str):
            self.require_valid_thumbnail_filename(new_thumbnail_filename)
```

Reply:
Nope,
There are cases in which `new_thumbnail_filename` can be `None`. So we can't do the validation directly. Last time I tried this resulted in backend errors in other files.
But the implementation I did can properly validate `new_thumbnail_filename`.

Reply:
Wait, then I'm bit confused, shouldn't it be 
```suggestion
        if isinstance(new_thumbnail_filename, str):
            self.require_valid_thumbnail_filename(new_thumbnail_filename)
```

Reply:
@vojtechjelinek 
This suggestion is reasonable iff. new_thumbnail would never be anything apart from string and None.
Suppose `new_thumbnail_filename` passes were an integer or something else. In that case, we aren't validating it which isn't good.
Or should I commit the suggestion under the assumption that `new_thumbnail_filename` will always be a string or None?

Reply:
Ah, right, then I think this should be fine. We want to validate whenever it is not `None`.
```suggestion
        if new_thumbnail_filename is not None:
            self.require_valid_thumbnail_filename(new_thumbnail_filename)
```

Reply:
yea, this one looks good.